merlin/utils.py

```diff
@@ -497,6 +498,31 @@ def contains_shell_ref(string):
     return False
 
 
+def dict_deep_merge(dict_a, dict_b, path=None):
+    """
+    This function recursively merges dict_b into dict_a. The built-in
```

Comment:
Would be useful to make it clear in the docstring that this only merges in new keys and will not update existing one (is there a need for an 'update' version of this that also updates existing keys?)

Reply:
I'll add that to the docstring here, thanks for pointing it out. In all the cases that I use this function I don't think I ever need to update existing keys. It's mainly used to help store all statuses that need to be read in/modified in one dict.